packages/go_router/lib/src/delegate.dart

```diff
@@ -147,8 +147,12 @@ class GoRouterDelegate extends RouterDelegate<RouteMatchList>
   }
 
   void _completeRouteMatch(Object? result, RouteMatchBase match) {
-    if (match is ImperativeRouteMatch) {
-      match.complete(result);
+    RouteMatchBase walker = match;
```

Comment:
i think it's more clear to walk the tree in _handlePopPage instead of _completeRouteMatch

Reply:
it's odd that the match here is not a final match, you still have to walk the tree.



but if you walk it right after assigning the match, it looks more clear?

```
 bool _handlePopPage(Route<Object?> route, Object? result) {
    final Page<Object?> page = route.settings as Page<Object?>;
    final RouteMatchBase match = _pageToRouteMatchBase[page]!;
+++
walk the tree 
+++
    return widget.onPopPageWithRouteMatch(route, result, match);
  }
``